packages/run-protocol/src/vpool-xyk-amm/estimate.js

```diff
@@ -0,0 +1,157 @@
+// @ts-check
+
+import { natSafeMath } from '@agoric/zoe/src/contractSupport/index.js';
+import { AmountMath } from '@agoric/ertp';
+import { BASIS_POINTS } from './constantProduct/defaults.js';
+
+/**
+ * @file support the UI in calculating how much to offer, taking slippage
+ * into account, and not requiring a round trip to the chain.
+ */
+
+const { ceilDivide, floorDivide } = natSafeMath;
+
+export const charge = (chargeBP, amount) => {
+  return AmountMath.make(
+    amount.brand,
+    ceilDivide(amount.value * chargeBP, BASIS_POINTS),
+  );
+};
+
+export const chargeFloor = (chargeBP, amount) => {
+  return AmountMath.make(
+    amount.brand,
+    floorDivide(amount.value * chargeBP, BASIS_POINTS),
+  );
+};
+
+const makeEstimator = (centralBrand, rates) => {
+  const { poolFeeBP, protocolFeeBP, slippageBP } = rates;
+
+  // Proceeds less slippage when specifying the amount to add to a single pool
+  function estimateSinglePoolProceeds(amountIn, brandOut, pools) {
+    const isCentralIn = amountIn.brand === centralBrand;
+    const pool = isCentralIn ? pools.get(brandOut) : pools.get(amountIn.brand);
+    const kPre =
+      pool.getSecondaryAmount().value * pool.getCentralAmount().value;
```

Comment:
Could this be pool.secondaryAmount.value instead (and similarly for `pool.centralAmount.value`)? That way the interface would match https://github.com/Agoric/agoric-sdk/blob/434a2408a9455b1ace2956eb541b7b99cb6fbc17/packages/run-protocol/src/vpool-xyk-amm/pool.js#L60-L63 which the UI has reference to.

Reply:
Done.